Source/Core/InputCommon/ControllerInterface/OctagonalMouseJail.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <array>
+#include <utility>
+
+#include "Core/HotkeyManager.h"
+
+// There is an example of how to use this code in DInputKeyboardAndMouse.cpp in the
+// UpdateCursorInput() function.
+namespace ciface
+{
+struct Point
+{
+  double x;
+  double y;
+};
+struct ExtendedWindowInfo
+{
+  void* handle;
+  double top;
+  double left;
+  double bottom;
+  double right;
+  double width;
+  double height;
+  double ratio;
+  Point center;
+};
+using PrimaryPrimitive = double;
+
+class OctagonalMouseJail
+{
+public:
+  OctagonalMouseJail();
+
+  // I'm not sure the singleton pattern is the best here. Making a
+  // global instance of the mouse jail on launch is probably the highest performance
+  // method of doing the same thing as this singleton. This jail gets made instantly
+  // anyway and is used continuously by DInputKeyboardMouse. The additional
+  // checks to see if it's already been intialized is probably just a waste of compute
+  // resources. This might be the easiest interface to use and read though. Just
+  // commenting in case performance of this becomes a problem. I haven't checked
+  // if the optimizer fixes this. I don't think it would, and I'm not proficient
+  // enough to check without wasting a bunch of time.
```

Comment:
A very rambly comment that doesn't actually give much info except "this is a singleton". I'd prefer if you just trashed it, this is extremely unlikely to cause any kind of performance issue.

Reply:
I would have said informative but to each their own.  I'll get rid of it